lua/caramba/orchestrator.lua

```
@@ -0,0 +1,278 @@
+local state = require('caramba.state')
+local utils = require('caramba.utils')
+local config = require('caramba.config')
+local planner = require('caramba.planner')
+local llm = require('caramba.llm')
+
+-- Short-lived cache of recently included related files to avoid repetition
+M._recent_related_files = {}
+local RELATED_TTL_SEC = 300
+
+local function now_sec()
+	return math.floor(vim.loop.now() / 1000)
```

Luacheck (1.2.0)

[warning] accessing undefined variable 'vim'

(W113)